crates/sqlexec/src/background_jobs.rs

```diff
@@ -0,0 +1,326 @@
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc, Mutex,
+    },
+    time::Duration,
+};
+
+use async_trait::async_trait;
+use tokio::{
+    sync::{mpsc, oneshot},
+    task::JoinHandle,
+    time::Instant,
+};
+
+use crate::errors::{ExecError, Result};
+
+/// Collection of background jobs.
+pub struct JobRunner {
+    send: mpsc::UnboundedSender<ChannelMessage>,
+    listen: Arc<Mutex<Option<JoinHandle<()>>>>,
```

Comment:
Not immediately obvious to me, why is this `Option`?

Reply:
So we can `take` the value out when `close()` is called. Basically to get ownership over `JoinHandle`.